### PR TITLE
code cleanup (+ minor refactor)

### PR DESCRIPTION
**Pull request by Ryddag**
## Description

- remove unused using
- remove redundant initializer
- convert if to switch when 'useful'
- made func() static when they can be static
- use link instead of foreach when possible
- enforce var usage
- reduce 'if' code nesting
- enforce usage of object initializer
- enforce property expression body when possible
- redundant code cleanup
- ...

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Celeste_Launcher_Gui/Config.cs [hunk 1]
```diff
@@ -76,19 +69,18 @@ public static UserConfig Load(string path)
...
                     // Get IPv4 address:
                     foreach (var ip in netInterface.GetIPProperties().UnicastAddresses)
                     {
-                        if (ip.Address.AddressFamily == AddressFamily.InterNetwork)
-                        {
-                            userConfig.MpSettings.PublicIp = ip.Address.ToString();
-                            return userConfig;
-                        }
+                        if (ip.Address.AddressFamily != AddressFamily.InterNetwork)
+                            continue;
```
Comment (martinmine) [inline Celeste_Launcher_Gui/Config.cs]: Avoid using `continue;` as this leaves another entry-point for this loop.

File: Celeste_Launcher_Gui/Pages/LoginPage.xaml.cs [hunk 2]
```diff
@@ -35,14 +26,14 @@ public LoginPage()
...
-            if (LegacyBootstrapper.UserConfig.LoginInfo.RememberMe == true && storedCredentials != null)
-            {
-                _logger.Information("User has already stored credentials before");
-                EmailInputField.InputContent = storedCredentials.Email;
-                PasswordInputField.PasswordInputBox.Password = "**********";
-                RememberPasswordOption.IsChecked = true;
-                AutoLoginOption.IsEnabled = true;
-            }
+            if (!LegacyBootstrapper.UserConfig.LoginInfo.RememberMe || storedCredentials == null)
+                return;
```
Comment (martinmine) [inline Celeste_Launcher_Gui/Pages/LoginPage.xaml.cs]: This leaves another exit-point in the code which makes it harder to follow, I prefer having the logic as it was before

File: Celeste_Launcher_Gui/Services/UpdateService.cs [hunk 3]
```diff
@@ -82,10 +82,7 @@ public static async Task<string> GetChangeLog()
                 var changelogFormatted = StripHtml(Markdown.ToHtml(changelogRaw))
                     .Replace("Full Changelog", string.Empty).Replace("Change Log", string.Empty);
 
-                if (!string.IsNullOrWhiteSpace(changelogFormatted))
-                    return changelogFormatted;
-
-                return Properties.Resources.UpdateServiceChangelogError;
+                return !string.IsNullOrWhiteSpace(changelogFormatted) ? changelogFormatted : Properties.Resources.UpdateServiceChangelogError;
```
Comment (martinmine) [inline Celeste_Launcher_Gui/Services/UpdateService.cs]: This statement is really difficult to read, I prefer having the old one as it is more readable.

File: Celeste_Launcher_Gui/Windows/MultiplayerSettings.xaml.cs [hunk 4]
```diff
@@ -147,16 +144,15 @@ private void LanConnectionTypeCheckBox_Checked(object sender, RoutedEventArgs e)
...
-                    if (ip.Address.AddressFamily == AddressFamily.InterNetwork)
-                    {
-                        RemoteIPField.InputContent = ip.Address.ToString();
-                        return;
-                    }
-                }
+                if (ip.Address.AddressFamily != AddressFamily.InterNetwork)
+                    continue;
+                RemoteIPField.InputContent = ip.Address.ToString();
+                return;
```
Comment (martinmine) [inline Celeste_Launcher_Gui/Windows/MultiplayerSettings.xaml.cs]: This code has now 3 exit points, please revert it to the old one as it is easier to follow.

File: Celeste_Launcher_Gui/Config.cs [hunk 5]
```diff
@@ -1,29 +1,22 @@
...
 using Celeste_Public_Api.Helpers;
 
-#endregion
-
 namespace Celeste_Launcher_Gui
 {
-    [SuppressMessage("ReSharper", "InconsistentNaming")]
     public enum GameLanguage
     {
+        // ReSharper disable InconsistentNaming
```
Comment (martinmine) [inline Celeste_Launcher_Gui/Config.cs]: I would personally prefer if you leave these out, I don't have ReSharper installed on my private machine (as it makes VS terribly slow) and neither am I intending to install it. Since ReSharper is outside both the C# standard and a product that costs money, I think we should leave this out.

**Comment by Ryddag** [comment]
Just to be clear none of those change have been "blindly apply", i checked all files one by one to find potential code improvement, some stuff are based on "ReSharper suggestion" indeed but only a few most are from "Code Analysis Warnings", the rest "by myself" and none of them have done without reason or "blindly".

I am not sure at all i "make the code worse" nesting has been reduce in many place who make it more readable and the 3/4 points where you think its worse i don't think the same for me its more readable like this but after from a person to another this can indeed differ.

About the the fact that i have add "entry/exit point" , in C the "SESE" rules indeed is important to avoid memory leak but in C# its not necessary at all to respect this since we have the possibility to use "using" or "finally" to avoid memory leak due to resource who was not closed/disposed before return.

Their is a lot of debate about "SESE" in c# (and so reducing nesting, since most of time the rules "SESE" involve more nesting), their are the one "for" and the one "against" but in the end everyone have right (at least in .Net).

My self i don't care about "SESE" rules i prefer to have pre-conditional check who "cancel" operation if false and less nesting than "force" me to respect "SESE" rules in an language where is not necessary. But after like i said upper this is an personal choice, and i respect yours, everyone have his own "code style", their is so many way to achieve the same goal why would we all do the same ^^

File: Celeste_Launcher_Gui/Config.cs (hunk 5 above)
Comment (Ryddag) [reply Celeste_Launcher_Gui/Config.cs]: I use it that why its here.
Since it just a "comment line" (and was already here before) i don't think it will arm if it stay.

**Comment by martinmine** [comment]
Looks like my bad morning mood caught up with my again, my apologizes, my comment was in no way intended that it makes the entire code-base worse in any way :) The overall change seems good, the only places I have commented on are the places I think should be changed. All in all, I think the changes you improve are combined an improvement.
The usage of `break;` and `continue;` are entries that I think of as code smells. Sure, it might make sense in low-level languages such as C, but this is C# where we should not be that concerned with low-level memory managed and high-performance applications where each CPU instruction would count. In languages like C#, and especially this launcher, the focus should be more on the actual domain and getting rid of clutter from what actually happens in the code. When hitting these kinds of statements, we (the developers) have to make another conceptual jump in our minds while reading the code.
I don't think nesting if statements as a concept is bad. But this depends on the code and what it does. If there are multiple nested conditions within a function that spans across several hundred lines, the code becomes hard to follow. One low-effort solution would be to reduce the nesting, but a proper solution would be to split the function up in smaller functions - and perhaps other classes. It might make sense in some cases to avoid the nesting, but in cases like those I have commented on, I don't think it improves readability. It entirely depends on the situation and that is partially what made me claim you applied them "blindly". It might seem really tempting to resolve a warning that your IDE is recommending you to do, but there are exceptions to those recommendations and I think introducing these keywords (`break`, `continue`, etc.) should be carefully justified - depending on the situation.
I want to emphasize that this is not based on any personal choice that I have but rather on what best-practices dictates (i.e. clean code principles) and how good quality code should look like. I am not making these statements up on subjective opinions but rather how maintainable code should be like.
I'm not sure what SESE rules are, could you provide me a link for that? Nor do I find anything on Google about it. Maybe you are mixing it up with something else?

File: Celeste_Launcher_Gui/Config.cs (hunk 5 above)
Comment (martinmine) [reply Celeste_Launcher_Gui/Config.cs]: This takes up two entire lines on my monitor which are comments. Comments should be informative text explaining _me_ what is going on in the code, and not to inform a tool to disable warnings (which may only benefit you and others who are using Resharper). That is my argument for leaving them out as it is clutter to those who don't use Resharper. Since Resharper is not a standard of C# or anything that comes out of the box from Visual Studio, my argument is still that you remove them.

**Comment by Ryddag** [comment]
https://blogs.msmvps.com/peterritchie/2008/03/07/single-entry-single-exit-should-it-still-be-applicable-in-object-oriented-languages/

**Comment by martinmine** [comment]
Ah, I understand. Like I said, it really depends on what you are doing, and in those cases it makes sense to go around that rule as the code becomes cleaner :)
